### PR TITLE
seq to list with scalafix

### LINKED ISSUE #1035: Use List[String] instead of Seq[String] internally

### PR DESCRIPTION
**Pull request by erikwj**
Fixes #1035

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,6 +1,9 @@
 import ReleaseTransformations._
 import microsites.ExtraMdFileConfig
 
+addCompilerPlugin("org.scalameta"  % "semanticdb-scalac" % "4.0.0" cross CrossVersion.full)
```
Comment (vkostyukov) [inline build.sbt]: Do we need this enabled going forward?

Comment (erikwj) [reply build.sbt]: it's needed for scalafix, I like to keep scalafix available, but I'm fine to clean up if you want to

Comment (erikwj) [reply build.sbt]: This did the trick for 90% of the code: 
```scalafix https://gist.githubusercontent.com/erikwj/1c576d0f78a095dc7ff2e14ebb71c8a9/raw/c139a6b92f9ee3a7b9edfa7dd6856dcff0010afa/SeqToList```

**Comment by vkostyukov** [comment]
@rpless @sergeykolbasov How do you feel about this?

File: build.sbt (hunk 1 above)
Comment (vkostyukov) [reply build.sbt]: Right. I just wasn't sure what's the reason to keep it around, until we have another rewrite. I'm voting for omitting it.

Comment (vkostyukov) [reply build.sbt]: Getting back to this. @erikwj, can we omit this for now (as it's no longer needed)?

**Comment by rpless** [comment]
I'm mostly fine with this, although there are 3 cases I'm not sure about:
- The `paths` method in Endpoint feels like it should be a Seq because there is an order
- The `accept` argument in `Negotiable` also feels like it should remain a sequential
- the `r` in `Trace.fromRoute` also is ordered

I'm kinda on the fence about changing these, but if people are okay with there being an implied order for the lists in these cases, then I can go along with that.

File: build.sbt (hunk 1 above)
Comment (erikwj) [reply build.sbt]: ok @vkostyukov will submit the changes

**Comment by rpless** [comment]
> Seq is that it could be substituted with mutable structures

That's a good point. Also didn't realize `List` extended `Seq`, thought it list weren't necessarily ordered, but I guess they are, so its not a problem then.

**Pull request merged**